File: clients/typescript/examples/tables-get-table-typed.ts
import { DataAPIClient, DataAPIDate } from "@datastax/astra-db-ts";

// Manually define the type of the table's schema and primary key
type TableSchema = {
  title: string;
  author: string;
  number_of_pages?: number | null | undefined;
  rating?: number | null | undefined;
  genres?: Set<string> | undefined;
  metadata?: Map<string, string> | undefined;
  is_checked_out?: boolean | null | undefined;
  due_date?: DataAPIDate | null | undefined;
};

type TablePrimaryKey = Pick<TableSchema, "title" | "author">;

// Get an existing table
const client = new DataAPIClient();
const database = client.db("**API_ENDPOINT**", {
  token: "**APPLICATION_TOKEN**",
});
const table = database.table<TableSchema, TablePrimaryKey>("**TABLE_NAME**");

File: clients/python/examples/tables-get-table-typed.py
from datetime import date
from typing import Dict, Optional, Set, TypedDict

from astrapy import DataAPIClient, Table


# Define the typing
class TableSchema(TypedDict):
    title: str
    author: str
    number_of_pages: int
    rating: float
    genres: Set[str]
    metadata: Dict[str, str]
    is_checked_out: bool
    due_date: Optional[date]


# Get an existing table
client = DataAPIClient()
database = client.get_database(
    "**API_ENDPOINT**", token="**APPLICATION_TOKEN**"
)
table: Table[TableSchema] = database.get_table(
    "**TABLE_NAME**", row_type=TableSchema
)
